plugins/modules/ndo_tenant_custom_qos_policy.py

```diff
@@ -0,0 +1,462 @@
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ to ]
+      dscp_target:
+        description:
+        - The DSCP target value.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ target ]
+      target_cos:
+        description:
+        - The target CoS to be driven based on the range of input values of DSCP coming into the fabric.
+        type: str
+        choices: [ background, best_effort, excellent_effort, critical_applications, video, voice, internetwork_control, network_control, unspecified ]
+      qos_priority:
+        description:
+        - The desired QoS class level to be used.
+        type: str
+        choices: [ level1, level2, level3, level4, level5, level6, unspecified ]
+        aliases: [ priority, prio ]
+  cos_mappings:
+    description:
+    - The CoS mappings of the Custom QoS Policy.
+    type: list
+    elements: dict
+    suboptions:
+      dot1p_from:
+        description:
+        - The Dot1P range starting value.
+        type: str
+        choices: [ background, best_effort, excellent_effort, critical_applications, video, voice, internetwork_control, network_control, unspecified ]
+        aliases: [ from ]
+      dot1p_to:
+        description:
+        - The Dot1P range ending value.
+        type: str
+        choices: [ background, best_effort, excellent_effort, critical_applications, video, voice, internetwork_control, network_control, unspecified ]
+        aliases: [ to ]
+      dscp_target:
+        description:
+        - The DSCP target value.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ target ]
+      target_cos:
+        description:
+        - The target CoS to be driven based on the range of input values of Dot1P coming into the fabric.
```

Comment:
not sure I fully understand this sentence

Reply:
I changed it to be simpler.